core/services/ocr2/plugins/ccip/config/config.go

```diff
@@ -42,8 +43,134 @@ func (c CommitPluginConfig) Encode() ([]byte, error) {
 
 // DynamicPriceGetterConfig specifies which configuration to use for getting the price of tokens (map keys).
 type DynamicPriceGetterConfig struct {
+	// Deprecated: use TokenPrices instead.
 	AggregatorPrices map[common.Address]AggregatorPriceConfig `json:"aggregatorPrices"`
-	StaticPrices     map[common.Address]StaticPriceConfig     `json:"staticPrices"`
+	// Deprecated: use TokenPrices instead.
+	StaticPrices map[common.Address]StaticPriceConfig `json:"staticPrices"`
+
+	TokenPrices []TokenPriceConfig `json:"tokenPrices"`
+}
+
+// IsDeprecated returns true if the config uses the deprecated fields.
+func (c *DynamicPriceGetterConfig) IsDeprecated() bool {
+	return len(c.AggregatorPrices) > 0 || len(c.StaticPrices) > 0
+}
+
+// TokenPriceConfig specifies the configuration for a token price.
+type TokenPriceConfig struct {
+	// TokenAddress is the address of the token on the chain that is deployed on.
+	TokenAddress common.Address `json:"tokenAddress"`
+	// ChainSelector is the chain selector of the chain that the token is deployed on (source or dest).
+	ChainSelector uint64 `json:"chainSelector,string"`
```

Comment:
Clever, this marshals it as a string?

Reply:
We should do this in 1.6. I don't think it'll break anything since it'll just be for logging purposes (JSON).